### PR TITLE
adds 404 page

### LINKED ISSUE #348: Frontend: 404 page
* **I'm submitting a ...**
  * bug report

* **What is the current behavior?**
When I visit https://rundfunk-mitbestimmen.de/foobar I see... nothing. Well, a blank screen. It's even HTTP 200 (OK).

* **What is the expected behavior?**
I should see: "404: Ooops, there is nothing here." and a proper HTTP error.

* **What is the motivation / use case for changing the behavior?**
Better user experience.

* **If the current behavior is a bug, please tell us (otherwise delete):**
  * Steps to reproduce
    1. visit https://rundfunk-mitbestimmen.de/foobar
    2. Enjoy the white!
  * The version of your browser, local postgresql installation and operating system
    Chromium Version 61.0.3163.100, Postgresql 9.6.5., Archlinux
  * A screenshot
selection_074

  * Stacktraces, related issues, suggestions how to fix
Browser console
 ```
vendor-8bcc13a0ebfe8ebb1711701d4b5e99e1.js:1083 Uncaught L {description: undefined, fileName: undefined, lineNumber: undefined, message: "/foobar", name: "UnrecognizedURLError", …}
```
Network:
selection_075

```
```

### PR DESCRIPTION
**Pull request by sigu**
fixes #348  

THis is my first PR to get comfortable with the workflow of this project

## clarifications
- do we write the javasript unit tests for emberjs?
- do we need a feature test covering this particular scenario?

### Screenshot
peek 2017-10-19 00-21

### CODE REVIEW

File: frontend/app/templates/404.hbs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="ui huge header">
+  Something not nice happend :-(
```
Comment (roschaefer) [inline frontend/app/templates/404.hbs]: I would suggest:
"Oops.."
"We could not find that page."
and in German:
"Ups.."
"Wir konnten die Seite nicht finden."

as HTML:
```handlebars
<div class="ui huge header">{{t '404.header'}}</div>
<p>{{t '404.sub-header'}}</p>
```

Find the translations in `/frontend/translations`

File: frontend/tests/unit/routes/404-test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { setupTest } from 'ember-mocha';
+
+describe('Unit | Route | 404', function() {
+  setupTest('route:404', {
+    // Specify the other units that are required for this test.
+    // needs: ['controller:foo']
```
Comment (roschaefer) [inline frontend/tests/unit/routes/404-test.js]: Slightly annoying, but the auto-generated unit test for routes don't have the analytics service included. Check the other unit tests for routes and simply copy&paste.

**Comment by sigu** [comment]
@roschaefer  it does not go to 404 automatically, forgot to check in the routes file 😂

**Comment by sigu** [comment]
Here is the updated translations. Thanks @roschaefer  for showing me the translations
peek 2017-10-19 15-20

**Comment by roschaefer** [comment]
Thanks! :tada: I'll investigate why our build server fails. Probably not related to this PR but to a dependency problem caused by updates.

To answer your questions:
1. Yes, we test the frontend. But in this particular case I find it negligible.
2. Same with full-stack tests. We need them if we make changes to the backend and the frontend. Which is not the case.

**Comment by roschaefer** [comment]
One more thing: If you work on your feature branch alone, I would suggest you to do a `git rebase` instead of a `git merge`:
1. Get updates: `git fetch origin`
2. Rebase interactively: `git rebase origin/master -i`
3. Push with force `git push theNameOfYourRemote -u myFeatureBranch --force`

**Comment by roschaefer** [comment]
I get a 200 HTTP (OK) response on localhost:4200/foobar. Is there a way to get a proper 404 HTTP error?

**Comment by roschaefer** [comment]
OK @sigu I'll merge this, it looks great. You don't have to rebase, let's do that on your next PR :wink: thank you so much for your contribution!

**Comment by sigu** [comment]
@roschaefer I think the `200` responses we are getting are because the javascript is successfully fetched from the server despite the route not matching

**Comment by roschaefer** [comment]
Yes, of course you're right. The ember frontend cannot send HTTP responses to itself..

**Pull request merged**